app/src/org/odk/collect/android/logic/ArchivedFormManagement.java

```diff
@@ -0,0 +1,102 @@
+package org.odk.collect.android.logic;
+
+import org.commcare.android.database.SqlStorage;
+import org.commcare.android.database.user.models.FormRecord;
+import org.commcare.android.javarosa.AndroidLogger;
+import org.commcare.android.tasks.FormRecordCleanupTask;
+import org.commcare.dalvik.application.CommCareApp;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.javarosa.core.services.Logger;
+import org.joda.time.DateTime;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Vector;
+
+/**
+ * Determines when saved forms should be purged.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class ArchivedFormManagement {
```

Comment:
The fix: reworking of the purge logic lifted from CommCareActivity. Now uses Joda date time for bulk of the computations (delegating through java's date parsing for the time being; we need to run a migration to remove this parse dependency)